### ISSUE TITLE
Process period not working

### ISSUE THREAD

**Issue by GDYendell** [dls-controls/pmacFilterControl#39]
Possibly only in the first received message, e.g.:

```
- Response sent: {"status":{"current_attenuation":15,"in_positions":[0,0,0,0],"last_processed_frame":-2,"last_received_frame":-2,"mode":2,"out_positions":[0,0,0,0],"pixel_count_thresholds":{"high1":2,"high2":2,"high3":2,"low1":2,"low2":2},"process_duration":97,"process_period":1669291223099739,"state":1,"time_since_last_message":0,"timeout":3.0,"version":"0.1"},"success":true}
...
- Response sent: {"status":{"current_attenuation":13,"in_positions":[0,0,0,0],"last_processed_frame":0,"last_received_frame":2,"mode":2,"out_positions":[0,0,0,0],"pixel_count_thresholds":{"high1":2,"high2":2,"high3":2,"low1":2,"low2":2},"process_duration":231,"process_period":1812954,"state":4,"time_since_last_message":2,"timeout":3.0,"version":"0.1"},"success":true}
```

Possibly caused by #34

**Comment by OCopping** [comment]
``"process_period":1669291223099739`` corresponds to the time between the Unix epoch and the time you posted that (i.e. 12:00pm Nov 24th 2022). For some reason on the first process loop, instead of subtracting the process start time it uses the epoch (i.e. 0). Will look into what could be causing this...

**Comment by OCopping** [comment]
This could be down to the fact that ``process_period_`` uses ``last_process_ts_`` to calculate the period. This doesn't appear to be assigned a timestamp value until after the ``process_period_`` is calculated, hence why it would take off zero.
https://github.com/dls-controls/pmacFilterControl/blob/e72a37555183bc9424b2968cf914de93ce47807a/src/pmacFilterControl.cpp#L104
https://github.com/dls-controls/pmacFilterControl/blob/e72a37555183bc9424b2968cf914de93ce47807a/src/pmacFilterControl.cpp#L584-L589

**Comment by OCopping** [comment]
This is an issue with the attenuation file having the option to close after a timeout. As the `last_process_ts_` is such a big number at program start, it times out instantly, and hence the attenuation file instantly is closed. A quick fix is to just set the timeout to an insanely large number, but this then negates the close on timeout option. A potential fix would be to set the `last_process_ts` to the time of the start of the scan.

**Comment by OCopping** [comment]
Closing as fixed with #55
